include/irr/asset/IVirtualTexture.h

```diff
@@ -423,9 +434,11 @@ class IVirtualTexture : public core::IReferenceCounted, public IVirtualTextureBa
             assert(_tilesPerDim<=MAX_TILES_PER_DIM);
             assert(_layers<=MAX_LAYERS);
         }
-        //TODO: this should also copy address allocator state, but from what i see our address allocators doesnt have copy ctors
-        IVTResidentStorage(core::smart_refctd_ptr<image_t>&& _image, uint32_t _layerShift, uint32_t _xmask) :
+        
+        IVTResidentStorage(core::smart_refctd_ptr<image_t>&& _image, const phys_pg_addr_alctr_t& _alctr, const void* _reservedSpc, uint32_t _layerShift, uint32_t _xmask) :
             image(std::move(_image)),
+            m_alctrReservedSpace(reinterpret_cast<uint8_t*>(impl::alloc_and_copy(phys_pg_addr_alctr_t::reserved_size(1u, _alctr.get_total_size(), 1u), _reservedSpc))),
```

Comment:
there' actually a special overload for `reserved_size` which takes the old allocator reference + new buffer size (in-case you want to grow or shrink)

example
https://github.com/buildaworldnet/IrrlichtBAW/blob/shader_pipeline/include/irr/core/alloc/PoolAddressAllocator.h#L164